packages/api/endpoints/granules.js

```diff
@@ -488,15 +489,32 @@ async function del(req, res) {
 async function get(req, res) {
   const {
     knex = await getKnexClient(),
+    collectionPgModel = new CollectionPgModel(),
   } = req.testContext || {};
   const { getRecoveryStatus } = req.query;
   const granuleId = req.params.granuleName;
+  const collectionId = req.params.collectionId;
+
   let granule;
+  let pgCollection;
   try {
-    granule = await getUniqueGranuleByGranuleId(knex, granuleId);
+    if (collectionId) {
+      pgCollection = await collectionPgModel.get(
+        knex, deconstructCollectionId(collectionId)
```

Comment:
Not this PR, but at some point we need to talk about *how much* of this we're doing for the sake of not changing the API.    I really think we should revisit the conversation about returning the internal ID, as this lookup could be eliminated at scale if we returned the cumulus_id for the collection and required it here. 

That said, there are .... user issues with that as that ID isn't promised (you can remove and re-add a collection and get a new id, for example.... although you can't actually due to db constraints).    

Something to think about, but non-blocking.